jellyhaj: init at version 0.2.0

RobinMarchart opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

added jellyhaj (and myself as a new maintainer)

jellyhaj is a terminal based client for Jellyfin that is modeled after the offical web client. It is focused on video media and can play them through libmpv.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

RobinMarchart committed:
maintainers: add RobinMarchart

RobinMarchart committed:
jellyhaj: init at version 0.2.0

teto reviewed (approved):
pretty cool. Worked for me

RobinMarchart commented:
v0.1.2 fixed another issue where non video media on the jellyfin server prevented the client from working

timschumi commented on pkgs/by-name/je/jellyfin-tui-rs/package.nix:
```diff
@@ -0,0 +1,51 @@
+  mpv-unwrapped,
+  sqlite,
+  versionCheckHook,
+  mpris ? stdenv.isLinux,
```
I think consistency-wise the naming here should be `useMpris` or `withMpris`.

phanirithvij commented on pkgs/by-name/je/jellyfin-tui-rs/package.nix:
```diff
@@ -0,0 +1,51 @@
+  src = fetchFromGitHub {
+    owner = "owo-uwu-nyaa";
+    repo = "jellyfin-tui-rs";
+    rev = "594fb385bbf2e245ce8f3650713219fa43772be2";
```
I suggest using `tag` for releases, and `rev` only for unstable commits (when not tagged yet but required).

RobinMarchart replied:
I switched it to `tag` and it seems to work.

teto commented on pkgs/by-name/je/jellyfin-tui-rs/package.nix:
```diff
@@ -0,0 +1,51 @@
+    license = lib.licenses.mit;
+    sourceProvenance = [ lib.sourceTypes.fromSource ];
+    mainProgram = "jellyfin-tui-rs";
+    homepage = "https://github.com/owo-uwu-nyaa/jellyfin-tui-rs";
```
I am not sure what are the requirements for new packages but what you mention is what refrained me from merging. 
Seems like there is a flake in the repo now, which can act as a workaround until the program matures enough to get merged ?

phanirithvij replied:
You can mark it as draft instead of closing it

RobinMarchart commented:
Updated the version to 0.2.0, renamed to a more unique name (jellyhaj)
I think I am now at a point where I implemented most low hanging fruits and it should be a lot more stable now

RobinMarchart commented:
added `__structuredAttrs=true` to satisfy the CI
